src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/AbstractPropertiesProvider.java

```diff
@@ -159,33 +159,31 @@ protected ItemMetadata addItemMetadata(IPropertiesCollector collector, String na
 	 * @return the hint name.
 	 */
 	protected String updateHint(IPropertiesCollector collector, PsiClass type) {
-		if (type == null) {
+		if (type == null
+				|| !type.isEnum()) {
```

Comment:
I understand that the code is better, but it looks like it is not releated with the fix, right?

Reply:
@angelozerr this line is related to the changes further down. it is not strictly necessary though. I can remove it if you want this to be the bare minimum required for a fix.

Reply:
@angelozerr moved this refactoring to a separate commit